handlers.go

```diff
@@ -1,85 +1,67 @@
-
 package main
+
 import (
-	"net/http"
-	"github.com/gorilla/mux"
-	"log"
 	"encoding/json"
+	"github.com/gorilla/mux"
 	"io/ioutil"
+	"net/http"
 )
 
-// TODO add message fields
-
-type StatusResponse struct {
-    Task_id string `json:"task_id"`
-    Status int `json:"status"`
-}
-
-type StartResponse struct {
-    Task_id string `json:"task_id"`
-}
-
-type StopResponse struct {
-    Task_id string `json:"task_id"`
-		ExitCode int `json:"exit_code"`
-}
-
-// path param task_id : identifier of task query
-// returns status of running service
+// StatusHandler returns status of running service.
 func StatusHandler(w http.ResponseWriter, r *http.Request) {
 
 	vars := mux.Vars(r)
 	task_id := vars["task_id"]
 
-	status := GetProcessStatus(task_id)
-	log.Printf("task_id=%s, status=%d", task_id, status)
+	res := GetProcessStatus(task_id)
 
-	responseBody := StatusResponse{task_id, status}
 	w.WriteHeader(200)
 	w.Header().Set("Content-Type", "application/json")
-	json.NewEncoder(w).Encode(responseBody)
+	json.NewEncoder(w).Encode(res)
 }
 
+// StopHandler handles the logic of a call to /stop to stop a process.
 func StopHandler(w http.ResponseWriter, r *http.Request) {
 	body, _ := ioutil.ReadAll(r.Body)
-	log.Printf("body=%s", body)
 
 	body_map := make(map[string]string)
 	json.Unmarshal(body, &body_map)
 
 	task_id := body_map["task_id"]
-	log.Printf("task_id=%s", task_id)
 
-	// TODO define and handle errors
-	output := StopProcess(task_id)
+	if task_id == "" {
+		w.WriteHeader(401)
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(ErrorMessage{"no task_id provided"})
+		return
+	}
 
-	log.Printf("task_id=%s, status=%s", task_id, output)
+	StopResponse := StopProcess(task_id)
 
-	responseBody := StartResponse{task_id}
 	w.WriteHeader(200)
 	w.Header().Set("Content-Type", "application/json")
-	json.NewEncoder(w).Encode(responseBody)
+	json.NewEncoder(w).Encode(StopResponse)
 
 }
 
-// return pid
+// StartHandler handles the logic of a call to /start to start a process.
 func StartHandler(w http.ResponseWriter, r *http.Request) {
 
 	body, _ := ioutil.ReadAll(r.Body)
-	log.Printf("body=%s", body)
 
 	body_map := make(map[string]string)
 	json.Unmarshal(body, &body_map)
 
 	command := body_map["command"]
+	if command == "" {
+		w.WriteHeader(401)
```

Comment:
I wonder why this is returning unauthorized while something like `StatusBadRequest` would be more suitable?